pandas/core/arrays/_mixins.py

```diff
@@ -204,6 +206,17 @@ def __setitem__(self, key, value):
     def _validate_setitem_value(self, value):
         return value
 
+    @overload
+    def __getitem__(self, key: int | np.integer) -> Any:
```

Comment:
not for this PR, but could/should we do a Generic think to tighten this Any?

Reply:
Possibly.  Should I create an issue for discussion?

Reply:
isn't tihs a PositionScalar ?